spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/AbstractJdbcConfiguration.java

```diff
@@ -109,4 +111,26 @@ public JdbcAggregateTemplate jdbcAggregateTemplate(ApplicationEventPublisher pub
 				converter, operations);
 		return new JdbcAggregateTemplate(publisher, context, converter, dataAccessStrategy);
 	}
+
+	/**
+	 * Register a {@link DataAccessStrategy} as a bean for reuse in the {@link JdbcAggregateOperations} and the
+	 * {@link RelationalConverter}.
+	 *
+	 * @param context
+	 * @param converter
+	 * @param operations
+	 * @return
+	 */
+	@Bean
+	public DataAccessStrategy getDataAccessStrategy(RelationalMappingContext context, JdbcConverter converter,
+			NamedParameterJdbcOperations operations) {
+
+		DefaultDataAccessStrategy defaultDataAccessStrategy = new DefaultDataAccessStrategy(new SqlGeneratorSource(context), context, converter, operations);
+
+		if (converter instanceof BasicJdbcConverter) {
```

Comment:
We need to reconsider this one. We should not alter beans in a `@Bean` method. The need for a relation resolver on converter level makes it obvious that we create a cycle during runtime.